### ISSUE TITLE
Database inconsistency - Cluster is null error

### ISSUE THREAD

**Issue by dcarr178** [orientechnologies/orientdb#3825]
(running server 2.0.5) In the log below, when deleting from cluster 15 I get a server error saying cluster 18 is null but there is no cluster 18 in my schema. This happened to me once before and the only thing I knew how to do is drop database and start over. How can I get the consistency back?

```
orientdb {db=river}> classes


CLASSES
----------------------------------------------+------------------------------------+------------+----------------+
 NAME                                         | SUPERCLASS                         | CLUSTERS   | RECORDS        |
----------------------------------------------+------------------------------------+------------+----------------+
 _studio                                      |                                    | 11         |              0 |
 E                                            |                                    | 10         |              0 |
 OFunction                                    |                                    | 6          |              0 |
 OIdentity                                    |                                    | -          |              0 |
 ORestricted                                  |                                    | -          |              0 |
 ORIDs                                        |                                    | 8          |              0 |
 ORole                                        | OIdentity                          | 4          |              3 |
 OSchedule                                    |                                    | 7          |              0 |
 OTriggered                                   |                                    | -          |              0 |
 OUser                                        | OIdentity                          | 5          |              4 |
 rE                                           | E                                  | 12         |              0 |
 rHasCategory                                 | rE                                 | 16         |              2 |
 tBrand                                       | tV                                 | 14         |              2 |
 tChannel                                     | tV                                 | 15         |              1 |
 tV                                           | V                                  | 13         |              0 |
 V                                            |                                    | 9          |              0 |
----------------------------------------------+------------------------------------+------------+----------------+
 TOTAL = 16                                                                                                   12 |
----------------------------------------------+------------------------------------+------------+----------------+
orientdb {db=river}> delete vertex tChannel

Error: com.orientechnologies.orient.core.exception.ODatabaseException: Error on retrieving record #18:0 (cluster: null)

Error: java.lang.IllegalArgumentException: Cluster 18 is null
```

**Comment by lvca** [comment]
Try calling `reload schema` from console. Seems the schema hasn't been updated.

**Comment by dcarr178** [comment]
I tried that a few times and closed down the console a few times and
restarted both it and the server but the problem is still there.

On Fri, Mar 27, 2015 at 2:27 PM, Luca Garulli <notifications@github.com>
wrote:

> Try calling reload schema from console. Seems the schema hasn't been
> updated.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/orientechnologies/orientdb/issues/3825#issuecomment-87078856>
> .
>



-- 
David Carr
801.231.4946

**Comment by lvca** [comment]
Could you share the database to debug where 18 comes out?

**Comment by dcarr178** [comment]
Once I removed my java hook from the config file and restarted OrientDB it worked fine. I think the underlying issue/concern is that any Java hook can completely torpedo the OrientDB server and there are no mechanisms to debug what is going on or what runtime problems are occurring. I was able to correlate this new problem to a recently-released java hook but the hook compiled with no errors and the errors only appeared at runtime.
